### PR TITLE
Bug fix for clientset generation

### PR DESCRIPTION
**Pull request by Sindica**
This fix kube controller manager leader election issue: https://github.com/futurewei-cloud/arktos/issues/538

Test done:
1. Local: print out rate limiter pointer for rest request
2. Run perf test 500 node with KCM leader election enabled.

### CODE REVIEW

File: cmd/workload-controller-manager/main.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func getConfig() (*controllerManagerConfig.Config, error) {
 	}
 
 	klog.Infof("Current workload controller port %d", workloadControllerPort)
-	klog.Infof("Rest client QPS %v, kubeconfig [%#v]", controllerManagerKubeConfig.GetConfig().QPS, controllerManagerKubeConfig.GetConfig())
```
Comment (chenqianfzh) [inline cmd/workload-controller-manager/main.go:95]: Nit: I see this bug is about QPS. But for the purpose of information, we need the controllerManagerKubeConfig.GetConfig(), not just QPS.

File: staging/src/k8s.io/client-go/kubernetes/clientset.go [hunk 2]
```diff
@@ -333,160 +333,163 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
...
+		configCopy := *currentConfig
 		if configCopy.RateLimiter == nil && configCopy.QPS > 0 {
 			configCopy.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(configCopy.QPS, configCopy.Burst)
 		}
+		configShallowCopy.AddConfig(&configCopy)
 	}
 	var cs Clientset
 	var err error
-	cs.admissionregistrationV1beta1, err = admissionregistrationv1beta1.NewForConfig(&configShallowCopy)
+	cs.admissionregistrationV1beta1, err = admissionregistrationv1beta1.NewForConfig(configShallowCopy)
```
Comment (chenqianfzh) [inline staging/src/k8s.io/client-go/kubernetes/clientset.go:347]: I am puzzled by this change. I see the signature admissionregistrationv1beta1.NewForConfig(c *rest.Config) in the code. Could you tell me how come this change works?

File: cmd/workload-controller-manager/main.go (hunk 1 above)
Comment (Sindica) [reply cmd/workload-controller-manager/main.go:95]: GetConfig has a lot of information. including cert data. It is best to print out non sensitive and useful information.

File: staging/src/k8s.io/client-go/kubernetes/clientset.go (hunk 2 above)
Comment (Sindica) [reply staging/src/k8s.io/client-go/kubernetes/clientset.go:347]: configShallowCopy has type *rest.Config

**Pull request merged**